DoorGate/ViewController.swift

```diff
@@ -24,34 +26,33 @@ class ViewController: UIViewController {
     
     override func viewDidLoad() {
         super.viewDidLoad()
-        
+
         //In Button Logic
-        inButton.rx.tap.scan(0) { (priorValue, _) in
-                return 1
-            }
-            .subscribe(onNext: { [unowned self] currentCount in
-                let count = Int(self.countLabel.text!)! + currentCount
-                self.logic(count: count)
-                })
+        inButton.rx.tap
+            .subscribe({ [unowned self] currentCount in
+                self.presenter.buttonTap(command: DoorCommand.goingIn)
+            })
             .disposed(by: disposeBag)
-       
         //Out Button Logic
-        outButton.rx.tap.scan(0) { (priorValue, _) in
-                return -1
-            }
-            .subscribe(onNext: { [unowned self] currentCount in
-                let count = Int(self.countLabel.text!)! + currentCount
-                self.logic(count: count)
+        outButton.rx.tap
+            .subscribe({ [unowned self] currentCount in
+                self.presenter.buttonTap(command: DoorCommand.goingOut)
+                })
+            .disposed(by: disposeBag)
+
+        presenter.doorEntity.state.asObservable()
```

Comment:
This seems weird to me; it seems you would want state to already be an observable on the presenter, rather than casting it to one in the viewcontroller?

Reply:
I have state being observed in the viewcontroller to update the UI

Reply:
agree with observing it in the view controller, but it seems the state should already be an observable in the presenter, not cast to one in the UI.